tests/enum_primary_key_tests.rs

```diff
@@ -0,0 +1,104 @@
+pub mod common;
+
+pub use common::{features::*, setup::*, TestContext};
+use pretty_assertions::assert_eq;
+use sea_orm::{
+    entity::prelude::*,
+    entity::*,
+    sea_query::{BinOper, Expr},
+    ActiveEnum as ActiveEnumTrait, DatabaseConnection,
+};
+
+#[sea_orm_macros::test]
+#[cfg(any(
+    feature = "sqlx-mysql",
+    feature = "sqlx-sqlite",
+    feature = "sqlx-postgres"
+))]
+async fn main() -> Result<(), DbErr> {
+    let ctx = TestContext::new("enum_primary_key_tests").await;
+    create_tables(&ctx.db).await?;
+    insert_teas(&ctx.db).await?;
+    ctx.delete().await;
+
+    Ok(())
+}
+
+pub async fn insert_teas(db: &DatabaseConnection) -> Result<(), DbErr> {
+    use teas::*;
+
+    let model = Model {
+        id: Tea::EverydayTea,
+        category: None,
+        color: None,
+    };
+
+    assert_eq!(
+        model,
+        ActiveModel {
+            id: Set(Tea::EverydayTea),
+            category: Set(None),
+            color: Set(None),
+            ..Default::default()
+        }
+        .insert(db)
+        .await?
+    );
+    assert_eq!(model, Entity::find().one(db).await?.unwrap());
+    assert_eq!(
+        model,
+        Entity::find()
+            .filter(Column::Id.is_not_null())
+            .filter(Column::Category.is_null())
+            .filter(Column::Color.is_null())
+            .one(db)
+            .await?
+            .unwrap()
+    );
+
+    let _ = ActiveModel {
+        category: Set(Some(Category::Big)),
+        color: Set(Some(Color::Black)),
+        ..model.into_active_model()
+    }
+    .save(db)
+    .await?;
```

Comment:
I don't understand the intention of this bit, and I think it SHOULD fail to insert instead of passing

Reply:
It does an update instead of insert.

```rs
// The `..model.into_active_model()` expands into

let _ = ActiveModel {
        category: Set(Some(Category::Big)),
        color: Set(Some(Color::Black)),
        id: Unchanged( ... )
    }
    .save(db)
    .await?;
```

Reply:
Right

Reply:
Can we also add an insert test case

Reply:
We can insert a row with duplicate ID. But I cannot assert the exact error, since it's database dependent. https://github.com/SeaQL/sea-orm/pull/1414/commits/66cffd4a2a3b5c1a45b26d46b6971eda0a3da8a6

Reply:
Yes, `is_err` would do

Reply:
Alright, good to go